docs/source/tutorials/qmd/07-error-correction.qmd

```diff
@@ -0,0 +1,623 @@
+tau2_mu = lsl.Var.new_param(10.0, distribution = prior_tau2_mu, name = "tau2_mu")
+
+# Define prior tau2_scale distributions using Inverse Gamma
+prior_tau2_scale = lsl.Dist(tfd.InverseGamma, concentration = a_var_scale, scale = b_var_scale)
+tau2_scale = lsl.Var.new_param(10.0, distribution = prior_tau2_scale, name = "tau2_scale")
+
+# Compute P-spline penalty matrix (2nd-order difference for smoothness)
+penalty = splines.pspline_penalty(d = n_param_splines, diff = 2)
+
+# Define penalty matrices for scale and mean functions
+penalty_scale = lsl.Var(penalty, name = "penalty_scale")
+penalty_mu = lsl.Var(penalty, name = "penalty_mu")
+
+# Compute eigenvalues of the penalty matrix
+evals = jax.numpy.linalg.eigvalsh(penalty)
+
+# Compute rank of the penalty matrix (number of positive eigenvalues)
+rank_scale = lsl.Value(jnp.sum(evals > 0.0), _name = "rank_scale")
+rank_mu = lsl.Value(jnp.sum(evals > 0.0), _name = "rank_mu")
+
+# Compute log determinant of the penalty matrix (ignoring zero eigenvalues)
+log_pdet = jnp.log(jnp.where(evals > 0.0, evals, 1.0)).sum()
+log_pdet_mu = lsl.Value(log_pdet, _name = "log_pdet_mu")
+log_pdet_scale = lsl.Value(log_pdet, _name = "log_pdet_scale")
+
+# Define prior distribution for spline coefficients (scale function)
+prior_coef_scale  = lsl.Dist(
+    MultivariateNormalDegenerate.from_penalty,
+    loc = jnp.zeros(shape = (n_param_splines,)),
+    var = tau2_scale,
+    pen = penalty_scale,
+    rank = rank_scale,
+    log_pdet = log_pdet_scale
+    )
+
+# Initialize spline coefficients for scale function
+start_value_scale = np.zeros(np.shape(penalty)[-1], np.float32)
+coef_scale = lsl.Var.new_param(start_value_scale, distribution = prior_coef_scale, name = "coef_scale")
+
+# Define prior distribution for spline coefficients (mean function)
+prior_coef_mu = lsl.Dist(
+    MultivariateNormalDegenerate.from_penalty,
+    loc = jnp.zeros(shape = (n_param_splines,)),
+    var = tau2_mu,
+    pen = penalty_mu,
+    rank = rank_mu,
+    log_pdet = log_pdet_mu
+    )
+
+# Initialize spline coefficients for mean function
+start_value_mu = np.zeros(np.shape(penalty)[-1], np.float32)
+coef_mu = lsl.Var.new_param(start_value_mu, distribution = prior_coef_mu, name = "coef_mu")
+```
+Then we can implement functions to calculate the the standard deviation and mean of our response, using {meth}`~.model.Var.new_calc`.
+
+```{python}
+#| label: std-and-mean
+# Compute the scale (standard deviation) of y using the spline model
+scale_of_y = lsl.Var.new_calc(
+    pred_fn,
+    beta0 = b0_scale,                       # Intercept for scale function
+    spline_coef = coef_scale,               # Spline coefficients for scale
+    basis_matrix = basis_matrix_var_scale,  # Basis matrix for scale
+    name = "scale_of_y"
+)
+
+# Transform scale_of_y to ensure positivity (exponential transformation)
+scale_of_y_transformed = lsl.Var.new_calc(jnp.exp, scale_of_y, name = "scale_of_y_transformed")
```

Comment:
I would turn the naming around here. This is the actual scale, not the transformed scale. The transformed scale is the log scale (the additive predictor).